sympy/calculus/util.py

```diff
@@ -253,6 +253,83 @@ def elm_domain(expr, intrvl):
         return _domain
 
 
+def periodicity(f, symbol):
+    """Returns the period of a given function, if possible.
+
+    Examples
+    ========
+    >>> from sympy import Symbol, sin, cos
+    >>> from sympy.calculus.util import periodicity
+    >>> x = Symbol('x')
+    >>> f = sin(x) + sin(2*x) + sin(3*x)
+    >>> periodicity(f, x)
+    2*pi
+    >>> periodicity(sin(x)*cos(x), x)
+    pi
+    """
+    from sympy import simplify, lcm_list
+    from sympy.functions.elementary.trigonometric import TrigonometricFunction
+
+    orig_f = f
+    f = simplify(orig_f)
```

Comment:
I used `simplify` instead of `trigsimp` because I wanted this functionality to be developed with a general case in mind (non-trigonometric functions). 
I think both give the same output in case of trigonometric functions.